main.go

```diff
@@ -269,148 +374,62 @@ func keygenCmd(c *cli.Context) error {
 			slog.Fatal(err)
 		}
 		publics[i] = pub
 	}
-	group := key.NewGroup(publics, threshold)
-	groupPath := path.Join(fs.Pwd(), gname)
-	if c.String("out") != "" {
-		groupPath = c.String("out")
-	}
-	if err := key.Save(groupPath, group, false); err != nil {
-		slog.Fatal(err)
-	}
-	slog.Printf("Group file written in %s. Distribute it to all the participants to start the DKG", groupPath)
-	return nil
-}
-
-func dkgCmd(c *cli.Context) error {
-	if !c.IsSet("group-init") {
-		slog.Fatal("dkg requires a group.toml file")
-	}
-	group := getGroup(c)
-	conf := contextToConfig(c)
-	if exit := resetBeaconDB(conf); exit {
-		os.Exit(0)
-	}
-	fs := key.NewFileStore(conf.ConfigFolder())
-	drand, err := core.NewDrand(fs, group, conf)
-	if err != nil {
-		slog.Fatal(err)
-	}
-	return runDkg(c, drand, fs)
-}
-
-func runDkg(c *cli.Context, d *core.Drand, ks key.Store) error {
-	var err error
-	if c.Bool("leader") {
-		err = d.StartDKG()
-	} else {
-		err = d.WaitDKG()
-	}
-	if err != nil {
-		slog.Fatal(err)
-	}
-	slog.Print("DKG setup finished!")
-
-	public, err := ks.LoadDistPublic()
-	if err != nil {
-		slog.Fatal(err)
-	}
-	dir := fs.Pwd()
-	p := path.Join(dir, dpublic)
-	key.Save(p, public, false)
-	slog.Print("distributed public key saved at ", p)
-	return nil
-}
-
-func beaconCmd(c *cli.Context) error {
-	conf := contextToConfig(c)
-	fs := key.NewFileStore(conf.ConfigFolder())
-	drand, err := core.LoadDrand(fs, conf)
-	if err != nil {
-		slog.Fatal(err)
-	}
-	drand.BeaconLoop()
-	return nil
-}
-
-func runCmd(c *cli.Context) error {
-	//x509 not available on windows: must run in insecure mode
-	if runtime.GOOS == "windows" && !c.Bool("insecure") {
-		slog.Fatal("TLS is not available on Windows, please run in insecure mode")
-	}
-	conf := contextToConfig(c)
-	fs := key.NewFileStore(conf.ConfigFolder())
-	var drand *core.Drand
-	var err error
-	if c.IsSet("group-init") {
-		group := getGroup(c)
-		if exit := resetBeaconDB(conf); exit {
-			os.Exit(0)
+	if c.IsSet("group") {
+		groupPath := c.String("group")
+		oldG := &key.Group{}
+		if err := key.Load(groupPath, oldG); err != nil {
+			slog.Fatal(err)
 		}
-		drand, err = core.NewDrand(fs, group, conf)
-		if err != nil {
+		group := oldG.MergeGroup(publics)
+		if err := key.Save(groupPath, group, false); err != nil {
 			slog.Fatal(err)
 		}
-		slog.Print("Starting the dkg first.")
-		runDkg(c, drand, fs)
+		slog.Printf("Group file updated can be found at %s. Run upgrade command to do the resharing.", groupPath)
 	} else {
-		_, errG := fs.LoadGroup()
-		_, errS := fs.LoadShare()
-		_, errD := fs.LoadDistPublic()
-		if errG != nil || errS != nil || errD != nil {
-			slog.Fatalf("The DKG has not been run before, please provide a group file to do the setup.")
+		config := contextToConfig(c)
+		group := key.NewGroup(publics, threshold, &key.DistPublic{})
+		groupPath := path.Join(config.ConfigFolder(), key.GroupFolderName)
+		if err := key.Load(groupPath, group); err == nil {
```

Comment:
Same here. I believe the whole group command should only be "static": it does not save or require any information except what's given in the command line.